sky/shell/ui/internals.h

```diff
@@ -46,6 +47,12 @@ class Internals
   mojo::ServiceProviderImpl service_provider_impl_;
   mojo::ServiceProviderPtr platform_service_provider_;
 
+  // Proxy for a ServiceProvider that exposes services from the application to
+  // the embedder.
+  mojo::ServiceProviderPtr application_service_provider_;
```

Comment:
There's no reason to hold on to this object.  We can just make it a local variable in the constructor that goes out of scope immediately.